File: src/main/java/com/bot/commands/StopCommand.java
package com.bot.commands;

import com.bot.voice.VoiceSendHandler;
import com.jagrosh.jdautilities.commandclient.Command;
import com.jagrosh.jdautilities.commandclient.CommandEvent;

public class StopCommand extends Command {

	public StopCommand() {
		this.name = "stop";
		this.arguments = "";
		this.help = "Stops stream and clears the current playlist";
	}

	@Override
	protected void execute(CommandEvent commandEvent) {
		// TODO: Add more permissions checks
		VoiceSendHandler handler = (VoiceSendHandler) commandEvent.getGuild().getAudioManager().getSendingHandler();
		if (handler == null) {
			commandEvent.reply(commandEvent.getClient().getWarning() + " I am not connected to a voice channel.");
		}
		else {
			handler.stop();
			commandEvent.reply(commandEvent.getClient().getSuccess() + " Stopped audio stream");
			commandEvent.getGuild().getAudioManager().closeAudioConnection();
		}
	}
}

File: src/main/java/com/bot/commands/traditional/voice/StopCommand.kt
package com.bot.commands.traditional.voice

import com.bot.commands.traditional.VoiceCommand
import com.jagrosh.jdautilities.command.CommandEvent
import datadog.trace.api.Trace

class StopCommand : VoiceCommand() {
    init {
        name = "stop"
        arguments = ""
        help = "Stops stream and clears the current playlist"
    }

    @Trace(operationName = "executeCommand", resourceName = "Stop")
    override fun executeCommand(commandEvent: CommandEvent) {
        val voiceConnection = guildVoiceProvider.getGuildVoiceConnection(commandEvent.guild)
        if (!voiceConnection.isConnected()) {
            commandEvent.reply(commandEvent.client.warning + " I am not connected to a voice channel.")
        } else {
            voiceConnection.cleanupPlayer()
            commandEvent.reply(commandEvent.client.success + " Stopped audio stream")
        }
    }
}